### PR TITLE
feature: VPC Egress Gateway

### PR DESCRIPTION
**Pull request by zhangzujian**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Features
- Bug fixes
- Docs
- Tests

## Which issue(s) this PR fixes

Fixes #(issue-number)

### CODE REVIEW

**Comment by patriziobassi** [comment]
would there be the possibility to have more than one egress gw? for instance we may want to select different egress gw IPs for different workload in order to filter on a firewall/router

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -319,6 +330,7 @@ func Run(ctx context.Context, config *Configuration) {
 	nodeInformer := informerFactory.Core().V1().Nodes()
 	serviceInformer := informerFactory.Core().V1().Services()
 	endpointInformer := informerFactory.Core().V1().Endpoints()
+	deploymentInformer := informerFactory.Apps().V1().Deployments()
```
Comment (oilbeater) [inline pkg/controller/controller.go:333]: 最好能过滤一下，避免不必要的内存开销

File: pkg/util/pod_routes.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/klog/v2"
+
+	"github.com/kubeovn/kube-ovn/pkg/request"
+)
+
+type PodRoutes map[string]map[string][]string
```
Comment (oilbeater) [inline pkg/util/pod_routes.go:12]: 定义一个新的结构？这个嵌套定义看起来太复杂了

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/controller/controller.go (hunk 1 above)
Comment (zhangzujian) [reply pkg/controller/controller.go]: done

File: pkg/util/pod_routes.go (hunk 2 above)
Comment (zhangzujian) [reply pkg/util/pod_routes.go]: 已修改

**Review by oilbeater — APPROVED** [review]

**Comment by zsxsoft** [comment]
any doc?

**Comment by zhangzujian** [comment]
> any doc?

Document will be added later soon.

**Comment by zhangzujian** [comment]
@patriziobassi @zsxsoft  Here is the [document](https://kubeovn.github.io/docs/v1.14.x/en/advance/vpc-egress-gateway/).

**Pull request merged**